search/utils.py

```diff
@@ -0,0 +1,23 @@
+import re
+from typing import Union
+
+import flask
```

Comment:
Problematic in terms of coupling to depends on `flask` in an `utils` module: `utils` often are a catch-all for code reuse, and many different modules require them and depend on them; the consequence is that all the modules will depend on `flask`, even if they are not related to the framework.

Suggestion to remove the dependency is to pass in `gateway_url` as an argument to `get_content`, so that the reference to the configuration is made in `populate` which is already retrieving the flask app and is already coupled strongly to blueprints and other concepts.

Reply:
I would prefer not to pass the gateway url if I can as it's a constant. What I can do is this: `import flask.current_app`. What do you think?

Reply:
Doesn't break the dependency, I'd still get the `gateway_url` from a place that has to know about Flask rather than from a utility module. Could have a class here:
```
class ApiGateway:
    def __init__(self, url):
        self._url = url
    def get(self, ...):
        ...
```
and then create a global API_GATEWAY object in the application bootstrap passing in the `gateway_url`; all so that blueprints can use it but this module remains reusable and unit testable.

Reply:
I don't think an object is necessary for a constant. I'll  remove this function and just use requests with a fully formed url.